### DISCUSSION TITLE
Using meson as the build system

### DISCUSSION THREAD

**Discussion by ndmlny-qs** [mkleehammer/pyodbc#1264] [Ideas]
Here are a few examples that show using a `pyproject.toml` file along with a `meson.build` file with no setup.py file.

https://github.com/numpy/numpy-user-dtypes

Here is the doucmentation to meson-python https://github.com/mesonbuild/meson-python. I've tried it out, and it works well. The plus side is that it removes the need for a setup.py file and meson is a familiar tool for C.

**Comment by mkleehammer** [comment]
I'll look into this.  I thought it would be good to keep the 3rd party requirements as minimal as possible, even though we do provide binary wheels.  But if it is pretty standard and useful, it might be worth it.

The other important consideration is how easy it is to use on GitHub's CI, which we use to build the binary wheels.  If it is provided with their standard builds, that would be helpful.  I'll check it out.

**Reply by ndmlny-qs** [reply]
It was suggested to me by some of the scientific python devs, and you can read about the rationale behind it in this read me https://github.com/scientific-python/spin/blob/main/README.md.
